test/compilable/testcolor.sh

```diff
@@ -0,0 +1,46 @@
+set -eo pipefail
+
+compare()
+{
+    local actual expected
+    actual="$1"
+    expected="$2"
+    if [ "$actual" != "$expected" ] ; then
+        echo "Expected: $expected"
+        echo "Actual:   $actual"
+        echo "$expected" | hexdump -C
+        echo "$actual" | hexdump -C
+        exit 1
+    fi
+}
+
+filterAnsi()
+{
+    # try to emulate the `strings` command
+    # sed on some OSes doesn't support direct hex byte in its input string
+    sed "s/$(printf "\x1b")\(\[[0-9;]*m\)/[\1/g" | sed "s/\[\[/[/g" | sed "s/m\[m/m/g" | tr -d "\n\r"
```

Comment:
Since we're OK with using bash, we can use extended literals. `s/$'\x1B''\(\[[0-9;]*m\)/[\1/g'`